scripts/versions.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import os
+import sys
+import argparse
+from subprocess import check_output
+from collections import namedtuple
+
+parser = argparse.ArgumentParser(description='Should be run from the root of the git repo. Given a source policy document, will spelunk the git commit history to find the most recent commit for every version number, and generate an instance of the document containing the policy text at that point in time.. The version number will be inserted into the filename (immediately prior to the file extension). Will overwrite any existing files blindly. Will not delete output from previous executions.')
+parser.add_argument('filepath',
+                    help='File path to the source policy file from which to generate the versions.')
+parser.add_argument('--destination', default='versions',
+                    help='Destination root to output the versioned files')
+parser.add_argument('--unversioned', default=None,
+                    help='Version number to use for unversioned commits. Defaults to None, which causes no file to be generated for policy commits without version information')
+args = parser.parse_args()
+
+Version = namedtuple('Version', ['commit', 'meta', 'text'])
+
+def list_commits(filepath):
+    command_output = check_output(['git',
+                                   'log',
+                                   '--reverse',
+                                   '--format=%H',
+                                   filepath]).decode('utf-8')
+    commits = [row for row in command_output.split('\n')
+               if row.strip()]
+    return commits
+
+
+def get_meta(text):
+    text = [line for line in text
+            if line.strip()]
+    index = [i for i, j in enumerate(text) if j == '---']
+    if len(index) >= 2 and index[0] == 0:
```

Comment:
I think this might have hurt my head less if you'd done something like
```
if len(index) != 2:
   return {}
[metastart, metaend] = index
if metastart != 0:
    return {}
```

Reply:
Replaced with a less-head-hurting version :)